catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/event/impl/SubscriptionImpl.java

```diff
@@ -63,4 +65,15 @@ public DeliveryMethod getDeliveryMethod() {
         return dm;
     }
 
+    @Override
+    public boolean isActive() {
+        return active;
+    }
+
+    @Override
+    public boolean checkActive() {
+        active = dm.ping();
```

Comment:
The original reason I did this way was that I didn't want to advertise that they could have received an event but it was being filtered out by an AccessPlugin.